### ISSUE TITLE
Whether it doesn't work for the full connection layer

### ISSUE THREAD

**Issue by wonderfulboy** [google-research/morph-net#56]
Thanks for your open souring. I'm confused about whether it doesn't work for the full connection layer?

**Comment by eladeban** [comment]
Fully connected layers are supported via the GroupLasso* regularizes. Does this answer your question?

**Comment by wonderfulboy** [comment]
> Fully connected layers are supported via the GroupLasso* regularizes. Does this answer your question?

thanks for your reply! i already  use the flop_regularizer.GroupLassoFlopsRegularizer, but the JSON file just has the convalution layer without fc layer. 

Besides, the README file says "use Gamma regularizer if the seed network has BatchNorm; use GroupLasso otherwise", now if my network architecture has both BN and fc, should i use  GroupLassoFlopsRegularizer or GammaFlopsRegularizer. 

thank you !

**Comment by eladeban** [comment]
It seems to me this is working as intended. 

Usually, the last layer (fully_connected or conv) has a **fixed** number of outputs, think of `num_labels`. For example if your network has the following structure:
```
net= conv(inputs...)
net= conv(net...)
...
net= conv(net...)
logits = fully_connected(net, num_labels)
```
You simply cannot regularize or learn the number of the output of the fully connected layer.

We will try to clarify the README to clarify this misconception.
